qt/python/mantidqt/widgets/codeeditor/completion.py

```diff
@@ -33,6 +33,8 @@
 import sys
 from keyword import kwlist as python_keywords
 from collections import namedtuple
+from lib2to3.pgen2.tokenize import detect_encoding
+from cStringIO import StringIO
```

Comment:
cStringIO is not in python 3, to solve move the import to your version checker and import from io for python 3

Reply:
True, but the pyhton3 StringIO doesn't work with detect_encoding. That's why the old one is used.

Reply:
Is there a method that works for python3?

Reply:
No, unfortunately. Though I think the issue is something that only appears in python2 anyway, so the changes should be able to be removed when we move over to python3 exclusively.

Reply:
In that case you will need to write this to work in python 3 to pass jenkins tests.

Reply:
There was a solution, I was just missing it. `detect_encoding` works just fine if you use `BytesIO` and encode the string first.

Reply:
Good job. Once the tests pass I'll approve.